zorg/buildbot/builders/ClangBuilder.py

```diff
@@ -494,103 +500,153 @@ def _getClangCMakeBuildFactory(
+                test_suite_cmd.extend(testsuite_flags)
+
+            # Only submit if a URL has been specified
+            if submitURL is not None:
+                if not isinstance(submitURL, list):
+                    submitURL = [submitURL]
+                for url in submitURL:
+                    test_suite_cmd.extend(['--submit', url])
+                # lnt runtest test-suite doesn't understand --no-machdep-info:
+                if testerName and not use_runtest_testsuite:
+                    test_suite_cmd.extend(['--no-machdep-info', testerName])
+
+            # CC and CXX are needed as env for build-tools
+            if vs and vs != "manual":
+                # VS environment requires some extra care.
+                f.addStep(SetProperty(
+                    command=builders_util.getVisualStudioEnvironment(vs, vs_target_arch),
+                    extract_fn=builders_util.extractVSEnvironment,
+                    env={'CC'  : cc, 'CXX' : cxx}))
+                test_suite_env = Property('vs_env')
+            else:
+                test_suite_env = copy.deepcopy(env)
+                test_suite_env['CC'] = cc
+                test_suite_env['CXX'] = cxx
+
+            # Steps to prepare, build and run LNT
+            f.addStep(ShellCommand(name='clean sandbox',
+                                command=['rm', '-rf', 'sandbox'],
+                                haltOnFailure=True,
+                                description='removing sandbox directory',
+                                workdir='test',
+                                env=env))
+            f.addStep(ShellCommand(name='recreate sandbox',
+                                command=['virtualenv', '--python=python3', 'sandbox'],
+                                haltOnFailure=True,
+                                description='recreating sandbox',
+                                workdir='test',
+                                env=env))
+            f.addStep(ShellCommand(name='setup lit',
+                                command=[python, lnt_setup, 'develop'],
+                                haltOnFailure=True,
+                                description='setting up LNT in sandbox',
+                                workdir='test/sandbox',
+                                env=env))
+            f.addStep(LitTestCommand(
+                                name='test-suite',
+                                command=test_suite_cmd,
+                                haltOnFailure=True,
+                                description=['running the test suite'],
+                                workdir='test/sandbox',
+                                logfiles={'configure.log'   : 'build/configure.log',
+                                            'build-tools.log' : 'build/build-tools.log',
+                                            'test.log'        : 'build/test.log',
+                                            'report.json'     : 'build/report.json'},
```

Comment:
What logs do we get from the cmake only test suite build? Is it just like a normal `ninja check-all`, stdio with the output, ending with a table of results?

Reply:
As most buildbots run these test for validation I have kept the log to mimic logs output generated by llvm-project's ninja check-all

If we run ninja check it will actually run `llvm-lit -sv .` which generates logs like we get in llvm-project ninja check-all. 

For detailed logs We should run `llvm-lit .` with sv to get logs like following:
 
`compile_time: 0.0323
exec_time: 0.0000
hash: "a787a4de011d3c645d10ccb5efa9df57"
link_time: 0.0561
size: 25984
size..bss: 8
size..comment: 151
size..data: 24
size..data.rel.ro: 568
size..dynamic: 528
size..dynstr: 412
size..dynsym: 456
size..eh_frame: 1652
size..eh_frame_hdr: 404
size..fini: 20
size..fini_array: 8
size..gcc_except_table: 124
size..gnu.hash: 28
size..gnu.version: 38
size..gnu.version_r: 144
size..got: 48
size..got.plt: 120
size..init: 24
size..init_array: 8
size..interp: 27
size..note.ABI-tag: 32
size..plt: 224
size..rela.dyn: 1800
size..rela.plt: 288
size..rodata: 61
size..text: 5128
**********`

Reply:
That part is fine, 99% of the time I'm looking at the "stdio" file anyway.

Is buildbot still able to parse the failures correctly and show the various failure types in the UI? (I forget if it parses the stdio or the json results file)